### PR TITLE
fix(jsdoc-to-markdown): correct module definition and update to v5

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- update module definition fixing errors like:
```bash
TSError: â¨¯ Unable to compile TypeScript:
index.ts:3:10 - error TS2339: Property 'render' does not exist on type
'typeof import("../node_modules/@types/jsdoc-to-markdown/index")'.

3 jsdoc2md.render({ files: 'src/*.js' }).then(...)
```
After this change the module is exported as regular CommonJS module
- update version to v5
- create backward compatible v4
- rewrite tests to match CommonJS module exports

https://github.com/jsdoc2md/jsdoc-to-markdown/blob/master/docs/API.md#jsdoc-to-markdown

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jsdoc2md/jsdoc-to-markdown/blob/master/docs/API.md#jsdoc-to-markdown
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
> Because this is a new definition

Oh, no, again Mr Bot :)

/cc @adamzerella

File: types/jsdoc-to-markdown/v4/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    source?: string;
+    /**
+     * The path to the jsdoc configuration file.
+     *  Default: path/to/jsdoc/conf.json.
+     */
+    configure?: string;
+}
+
+export class JsdocToMarkdown {
```
Comment (amcasey) [inline types/jsdoc-to-markdown/v4/index.d.ts]: I feel like I'm missing something obvious, but this class wasn't exported in v4 either, was it?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/jsdoc-to-markdown/v4/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/jsdoc-to-markdown/v4/index.d.ts]: @amcasey amended, yes, left the v4 without a revision as I was not sure what's the correct approach. Thx for reviewing!

Comment (amcasey) [reply types/jsdoc-to-markdown/v4/index.d.ts]: @peterblazejewicz We usually allow breaking changes without a version bump if the existing state is incorrect (e.g. won't actually work at runtime), so I think it makes sense to clean up the old one too.

**Review by amcasey — APPROVED** [review]

**Pull request merged**